### PR TITLE
8540 add github action

### PR DESCRIPTION
**Pull request by nglelinh**
@jayay 
please add the following environment value for testing the deploy action:

SVNURL: https://nglelinh@svn.code.sf.net/p/onofficewpplugin/code
SVN_USER: nglelinh
SVN_PASSWORD: XXXX (hidden)

I am using https://sourceforge.net/ for a free public SVN repository. You can also use this account to login and view the result.
PS: please change the extension of release.yml.bk to release.yml for testing the deploy action
This is my testing on another repo: https://snipboard.io/OGC4f3.jpg

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    - name: Checkout from GIT
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+
+    - name: Setup PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ matrix.php-versions }}
+        extensions: mbstring, mysqli, intl #optional, setup extensions
```
Comment (jayay) [inline .github/workflows/main.yml:34]: Does it make sense to mention `curl` as a required extension?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+name: Testing
+
+# Controls when the action will run.
+on:
```
Comment (jayay) [inline .github/workflows/main.yml:4]: Travis CI used to also test the `master` branch once a week by cron. As far as I know that's possible here as well. I thin that should be added here as well.

Comment (jayay) [reply .github/workflows/main.yml:4]: And every branch should be included, right?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      matrix:
+        operating-system: [ubuntu-latest]
+        php-versions: ['7.0', '7.1', '7.2', '7.3']
+    name: Unittest on ${{ matrix.operating-system }} with PHP version ${{ matrix.php-versions }}
+    steps:
+    - name: Set up MySQL
+      uses: mirromutth/mysql-action@v1.1
+      with:
+        mysql version: '5.7'
+        mysql root password: '123456'
```
Comment (jayay) [inline .github/workflows/main.yml:23]: Is it ok to use a public password here? I do think it is, but is there any documentation on this?

File: .github/workflows/release.yml.bk [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    name: Deployment to WP SVN (tag only)
+    steps:
+    - name: Checkout from GIT
+      uses: actions/checkout@v2
+
+    - name: Run svn commit script
+      run: sh bin/commit-svn
```
Comment (jayay) [inline .github/workflows/release.yml.bk:20]: `bin/commit-svn` requires some arguments: `$TRAVIS_TAG --stable`. Is that doable here?

**Review by jayay — CHANGES_REQUESTED** [review]
Looks pretty good already. I'm missing a cronjob for the tests once a week. Also, is there a way to test the deployment without creating a tag? The `commit-svn` script needs the tag name as an argument.
I put your credentials into the repo settings.

File: .github/workflows/main.yml (hunk 3 above)
Comment (nglelinh) [reply .github/workflows/main.yml:23]: we can transfer a secret value from env variable

File: .github/workflows/main.yml (hunk 1 above)
Comment (nglelinh) [reply .github/workflows/main.yml:34]: These extensions are required for running composer install and testing. It's running without `curl`. But yes, we can add extension `curl`

**Comment by nglelinh** [comment]
@jayay I fixed your comments, please help to review again
For testing, I think you can test by creating a new private repo and setup tag on that repo. Unfortunately, could not find any better way to test the tag

**Comment by jayay** [comment]
The unit tests look good. In order to enable the release mechanism, I guess `.github/workflows/release.yml.bk` needs to be renamed into `.github/workflows/release.yml`?
The credentials were set up in the repository settings. The next steps to test a release would be to remove the `.travis.yml`, rename `release.yml.bk` and to create a tag, is that correct?

**Comment by nglelinh** [comment]
> The unit tests look good. In order to enable the release mechanism, I guess `.github/workflows/release.yml.bk` needs to be renamed into `.github/workflows/release.yml`?
> The credentials were set up in the repository settings. The next steps to test a release would be to remove the `.travis.yml`, rename `release.yml.bk` and to create a tag, is that correct?

Yes, that's right. I didn't want to mix the 2 processes together so I changed the file extension.

**Comment by jayay** [comment]
Merged in f1cb1ed989d898d6427ac1b8b97e4548a7cbc0cc. Thank you!

**Pull request closed without merging**